services/backend/api/geocoder.py

```diff
@@ -0,0 +1,97 @@
+from typing import List
+from requests.api import request
+from .models import Location
+import requests
+import os
+from functools import lru_cache
+import logging
+
+logger = logging.getLogger(__file__)
+
+
+API_KEY = os.environ.get("GEOCODER_API_KEY")
+if not API_KEY:
+    logger.warn("Geocoder API key env var not set.")
+
+
+class GeocoderError(Exception):
+    pass
+
+
+def parse_location_string(location_string: str) -> Location:
+    """Parses a string formatted like `{lat}+{lon}` into a Location object
+
+    e.g. `34.05513+-118.25703` -> Location(lat: 4.05513, lon: -118.25703)
+
+    Parameters
+    ----------
+    location_string: str
+        A string formatted as specified above
+
+    Returns
+    -------
+    Location
+
+    Raises
+    ------
+    ValueError
+        If entity can't be parsed
+    """
+    # This would arguable be simpler with regex but I don't care to bother with that right now.
+    if "+" in location_string:
+        split_string = location_string.split("+")
+        if len(split_string) == 2:
+            lat, lon = split_string
+            return Location(lat=float(lat), lon=float(lon))
+
+    raise ValueError(f"Unable to parse {location_string}. Expected format 'lat+lon'")
+
+
+@lru_cache(maxsize=256)
```

Comment:
Adding caching seemed like a handy little trick. Suppose if we really wanted to get fancy we could add a full DB cache that would persist between runs.